### ISSUE TITLE
Annotate positional parameter names for literal arguments in IDEs

### ISSUE THREAD

**Issue by yjbanov** [dart-lang/sdk#30479]
Imagine the following function:

```dart
createDate(int year, int month, int day, bool isUtc);
```

Here's an example call site:

```dart
var month = 8;
createDate(2017, month, 17, true);
```

Because I used an identifier for month, it is clear what is passed and where. However, for literal values it is not clear what those arguments mean. It would be great if the IDE could annotate the arguments like this:

```dart
var month = 8;
createDate(year: 2017, month, day: 17, isUtc: true);
```

This is similar to how [Dart Code annotates closing parens](https://twitter.com/flutterio/status/898267497824002048). Here's an example of how IntelliJ annotates arguments for Go:

<img width="428" alt="screen shot 2017-08-17 at 2 56 00 pm" src="https://user-images.githubusercontent.com/211513/29437939-90098444-8368-11e7-9db3-d768289a7e79.png">

Where the function signature is:

```go
func createDate(year int, month int, day int, isUtc bool) {}
```

**Comment by zoechi** [comment]
Wouldn't this make it confusing when there are also named parameters?

**Comment by bwilkerson** [comment]
@lukechurch Has there been any indication in UX studies that this is an issue? (I'm guessing that most of the studies have used the Flutter code base, which already makes heavy use of named parameters, so this won't have shown up.)

**Comment by devoncarew** [comment]
I've used this feature in the IntelliJ Java tooling, and found it useful. That is in a language w/o named params however. In a language with them, I wonder if inserting synthetic text for the names for positional params would be too noisy?

It looks like (from what you say above) that the golang integration just uses the inserted text for literals? So, restricts the usage to places that could benefit the most from it.

**Comment by yjbanov** [comment]
Yes, I have only seen them used for literals. Note that golang struct constructors use named parameters, but `func`s don't. So this feature only applies to `func`s.

**Comment by helin24** [comment]
This has been accomplished with inlay hints (https://github.com/flutter/dart-intellij-third-party/issues/159)
